### PR TITLE
[hma] Add a settings pane for getting index last modified and rebuilding on demand

### PR DESCRIPTION
**Pull request by schatten**
Summary
---------
With banks, indexing runs on a pre-determined frequency. This may not be fast enough for our users when they are demoing something or need to test changes out.

This shows the last_modified times of the current set of indexes on the 'settings' portion of the HMA UI and allows a user to trigger an async rebuild of the index.

Test Plan
---------
mypy, black, py.test on local.

**Screenshot 1: the new UI pane with the rebuild button**
Screen Shot 2021-12-13 at 22 37 58

**Screenshot 2: logs of indexer when the rebuild button was clicked**
Screen Shot 2021-12-13 at 22 38 10

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/indexers/s3_indexers.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/indexers/s3_indexers.py:39]: (nonblocking) We haven't been consistent about this being a variable vs just locally defining it. It used to be something defined in terraform...

Comment (BarrettOlson) [reply hasher-matcher-actioner/hmalib/indexers/s3_indexers.py:39]: nevermind, looks like this is a step in the right direction regardless

File: hasher-matcher-actioner/hmalib/indexers/s3_indexers.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/indexers/s3_indexers.py:68]: How is "LastModified" stored? depending on the timestamp counting up wouldn't this actually be the most recently modified?

File: hasher-matcher-actioner/hmalib/lambdas/api/indexes.py
Comment (BarrettOlson) [inline hasher-matcher-actioner/hmalib/lambdas/api/indexes.py:40]: hmm, 'latest' or 'oldest' am I miss understanding these as contradictory.

**Review by BarrettOlson — APPROVED** [review]
Approving to unblock. Thanks
two asides though
1) can you confirm intended behavior is correct e.g. "oldest" vs "latest"

2) to me this index button is really nice to have but it could just be a button/card on a existing settings or bank page instead of its own dedicated one.

File: hasher-matcher-actioner/hmalib/indexers/s3_indexers.py
Comment (schatten) [reply hasher-matcher-actioner/hmalib/indexers/s3_indexers.py:68]: Yes. This is the most recently modified timestamp.

I am not accounting for a case where one index is built but the other is not.

**Comment by schatten** [comment]
>     1. can you confirm intended behavior is correct e.g. "oldest" vs "latest"

Fixed. It is latest. Not oldest. I started with oldest but quickly realized this would be really hard to get right. So switched everywhere but the method name. This is now addressed.

>     2. to me this index button is really nice to have but it could just be a button/card on a existing settings or bank page instead of its own dedicated one.

I agree. I struggled for a bit, but I could not figure out a place where this would fit better. Maybe a global settings page, but we don't have one. The equivalent settings for actions and action rules have their own page, so decided to go down this path, but am not beholden to it. 

Happy to discuss and move elsewhere once this lands.

**Pull request merged**
